receiver/hostmetricsreceiver/internal/scraper/psiscraper/psi_scraper.go

```diff
@@ -0,0 +1,191 @@
+	now := pcommon.NewTimestampFromTime(time.Now())
+	var errors scrapererror.ScrapeErrors
+
+	// Get root path
+	rootPath := s.config.RootPath
+	if rootPath == "" {
+		rootPath = "/"
+	}
+
+	cpuStats, err := readPSIFile(filepath.Join(rootPath, cpuFile))
+	if err != nil {
+		errors.AddPartial(4, err)
+	} else {
+		s.mb.RecordSystemPsiCPUSomeAvg10DataPoint(now, cpuStats.some.avg10)
+		s.mb.RecordSystemPsiCPUSomeAvg60DataPoint(now, cpuStats.some.avg60)
+		s.mb.RecordSystemPsiCPUSomeAvg300DataPoint(now, cpuStats.some.avg300)
+		s.mb.RecordSystemPsiCPUSomeTotalDataPoint(now, cpuStats.some.total)
+	}
+
+	memStats, err := readPSIFile(filepath.Join(rootPath, memoryFile))
+	if err != nil {
+		errors.AddPartial(8, err)
+	} else {
+		s.mb.RecordSystemPsiMemorySomeAvg10DataPoint(now, memStats.some.avg10)
+		s.mb.RecordSystemPsiMemorySomeAvg60DataPoint(now, memStats.some.avg60)
+		s.mb.RecordSystemPsiMemorySomeAvg300DataPoint(now, memStats.some.avg300)
+		s.mb.RecordSystemPsiMemorySomeTotalDataPoint(now, memStats.some.total)
+		s.mb.RecordSystemPsiMemoryFullAvg10DataPoint(now, memStats.full.avg10)
+		s.mb.RecordSystemPsiMemoryFullAvg60DataPoint(now, memStats.full.avg60)
+		s.mb.RecordSystemPsiMemoryFullAvg300DataPoint(now, memStats.full.avg300)
+		s.mb.RecordSystemPsiMemoryFullTotalDataPoint(now, memStats.full.total)
+	}
+
+	ioStats, err := readPSIFile(filepath.Join(rootPath, ioFile))
+	if err != nil {
+		errors.AddPartial(8, err)
+	} else {
+		s.mb.RecordSystemPsiIoSomeAvg10DataPoint(now, ioStats.some.avg10)
+		s.mb.RecordSystemPsiIoSomeAvg60DataPoint(now, ioStats.some.avg60)
+		s.mb.RecordSystemPsiIoSomeAvg300DataPoint(now, ioStats.some.avg300)
+		s.mb.RecordSystemPsiIoSomeTotalDataPoint(now, ioStats.some.total)
+		s.mb.RecordSystemPsiIoFullAvg10DataPoint(now, ioStats.full.avg10)
+		s.mb.RecordSystemPsiIoFullAvg60DataPoint(now, ioStats.full.avg60)
+		s.mb.RecordSystemPsiIoFullAvg300DataPoint(now, ioStats.full.avg300)
+		s.mb.RecordSystemPsiIoFullTotalDataPoint(now, ioStats.full.total)
+	}
+
+	return s.mb.Emit(), errors.Combine()
+}
+
+func readPSIFile(path string) (*psiStats, error) {
+	content, err := os.ReadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read PSI file %s: %w", path, err)
+	}
+
+	return parsePSI(string(content))
+}
+
+// parsePSI parses PSI file content
+// Format:
+// some avg10=0.00 avg60=0.00 avg300=0.00 total=0
+// full avg10=0.00 avg60=0.00 avg300=0.00 total=0
+func parsePSI(content string) (*psiStats, error) {
+	stats := &psiStats{}
+	lines := strings.Split(strings.TrimSpace(content), "\n")
+
+	for _, line := range lines {
+		line = strings.TrimSpace(line)
+		if line == "" {
+			continue
+		}
+
+		parts := strings.Fields(line)
+		if len(parts) < 5 {
+			return nil, fmt.Errorf("invalid PSI line format: %s", line)
+		}
+
+		pressureType := parts[0]
+		var avg10, avg60, avg300 float64
+		var total int64
+		var err error
+
+		// Parse avg10=X.XX
+		if strings.HasPrefix(parts[1], "avg10=") {
```

Comment:
Shouldn't `part[1]` _always_ start with `avg10=`?  If the prefix isn't present maybe we should fail here.  If any of the parts doesn't start with the expected prefix, we'll almost certainly not have any of the other parts with the expected prefix.  In these cases we'd just set all the values to 0 silently without any warning or error message.  I think the most likely scenario for this is if Linux changes the format of the PSI proc interface, by perhaps adding a different aggregation or time period.